Plugins/BuildServerIntegration/AppVeyorIntegration/AppVeyorAdapter.cs

```diff
@@ -218,49 +208,20 @@ private IEnumerable<AppVeyorBuildInfo> QueryBuildsResults(Project project)
                         var myBuilds = builds.Children();
                         var baseApiUrl = ApiBaseUrl + project.Id;
                         var baseWebUrl = WebSiteUrl + "/project/" + project.Id + "/build/";
-                        var isGitHubRepository = jobDescription["project"]["repositoryType"].ToObject<string>() ==
-                                                    "gitHub";
-                        var repoName = jobDescription["project"]["repositoryName"].ToObject<string>();
 
                         var buildDetails = new List<AppVeyorBuildInfo>();
                         foreach (var b in myBuilds)
                         {
-                            string commitSha1 = null;
-                            var pullRequestId = b["pullRequestId"];
-                            if (isGitHubRepository && pullRequestId != null)
+                            ObjectId objectId;
+                            if (!ObjectId.TryParse(b["commitId"].ToObject<string>(), out objectId) || !_isCommitInRevisionGrid(objectId))
                             {
-                                if (!_shouldDisplayGitHubPullRequestBuilds)
+                                if (!ObjectId.TryParse(b["pullRequestHeadCommitId"].ToObject<string>(), out objectId) || !_isCommitInRevisionGrid(objectId))
```

Comment:
can we simplify it?
e.g. something like

```
if (!IsCommitInRevisionGrid(b["commitId"]) && !IsCommitInRevisionGrid(b["pullRequestHeadCommitId"])) 
{
    continue;
} 

bool IsCommitInRevisionGrid(string commitId)
{
    ObjectId objectId;
    return !ObjectId.TryParse(commitId.ToObject<string>(), out objectId) || !_isCommitInRevisionGrid(objectId)
}
```

Reply:
Perhaps we could simplify a little but not like that because we need the `objectid` out value.

I will have a look.

There is another solution by not verifying if it's in the revision grid but I don't know which one has the best performance (and it's difficult to measure it)